index.js

```diff
@@ -26,19 +30,29 @@ function parse (text, reviver, options) {
     return obj
   }
 
-  // Check original string for potential exploit
-  if (!text.match(suspectRx)) {
-    return obj
+  if (protoAction !== 'ignore' && constructorAction !== 'ignore') {
+    if (!text.match(suspectProtoRx) && !text.match(suspectConstructorRx)) {
```

Comment:
I don't have any data to back this up, but maybe the `test` function is faster than the `match` function:

```suggestion
    if (suspectProtoRx.test(text) === false && suspectConstructorRx.test(text) === false) {
```

Reply:
Yes, it should be faster, but if in the future we'll need to use the `g` flag, we'll need to refactor this.